In [155]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [156]:
cei_16 = pd.read_csv("CEI_16.csv")
cd_17 = pd.read_csv("CD_17.csv", sep=";")
grad_16 = pd.read_csv("graduados_16.csv")

### Forma general de las cosas

In [157]:
cei_16.head(2)

N° ORDEN              NOMBRE  Padrón        DNI
0         1  AAS, PETTER FOSSAN  100346   29086232
1         2      ABABOU, KHALIL   97708  RA1057904

In [158]:
cd_17.head(2)

N° ORDEN               NOMBRE  Padrón       DNI
0         1  ABAD, ARIEL IGNACIO   86213  32166584
1         2        ABADI, BETINA   93386  35902423

In [159]:
grad_16.head(2)

N° ORDEN                     NOMBRE  Padrón       DNI
0        30   ABITANTE, CARLOS ALBERTO   79742  27177606
1        50  ACEVEDO, FERNANDO GUSTAVO   78374  27822506

## Chequeo de tipos

In [160]:
cd_17.dtypes

N° ORDEN     int64
NOMBRE      object
Padrón       int64
DNI          int64
dtype: object

In [161]:
cei_16.dtypes

N° ORDEN     int64
NOMBRE      object
Padrón      object
DNI         object
dtype: object

Por algún motivo el del CEI no agarra bien a todos los padrones como números. Esto causaría problemas. Vamos a chequear por qué.

In [162]:
pad = pd.to_numeric(cei_16["Padrón"], errors="force")

In [163]:
cei_16[pad.isnull()]

N° ORDEN                          NOMBRE Padrón       DNI
9780      9781  SALAZAR VARGAS, CARLOS ROBERTO  B0239  93890243

Hay **1 padrón fallado**. Este nos hacía que el padrón fuera una string en cei_16, lo cual nos arruinaría un join por padrón. Haremos un append y quitaremos repetidos luego.

Pasamos todo en cd a string para que sea más fácil chequear duplicados y comparar.

In [164]:
cd_17["Padrón"] = cd_17["Padrón"].apply(str)
cd_17["DNI"] = cd_17["DNI"].apply(str)
grad_16["Padrón"] = grad_16["Padrón"].apply(str)
grad_16["DNI"] = grad_16["DNI"].apply(str)

## Chequeo de duplicados en cada uno.

### Graduados

In [165]:
grad_16.duplicated(subset="DNI").sum()

0

### CD

In [166]:
cd_17.duplicated(subset="DNI").sum()

18

In [167]:
cd_17[cd_17.duplicated(subset="DNI", keep=False)]

N° ORDEN                            NOMBRE  Padrón       DNI
265        266    ALVAREZ WINDEY, JUAN SEBASTIÁN   95242  36755372
266        267    ALVAREZ WINDEY, JUAN SEBASTIÁN   95242  36755372
1024      1025            BORDONI, LUCAS ENRIQUE   96033  35425609
1025      1026            BORDONI, LUCAS ENRIQUE   96033  35425609
1259      1260          CABRERA, MARÍA FLORENCIA   97864  39322189
1260      1261          CABRERA, MARÍA FLORENCIA   97864  39322189
1957      1958               CONTINANZA, NICOLAS   97576  38616453
1958      1959               CONTINANZA, NICOLAS   97576  38616453
2167      2168              DAHAB, MOISES MICAEL   97520  38995013
2168      2169              DAHAB, MOISES MICAEL   97520  38995013
2960      2961    FERREIRO CEDONES, JAZMIN SOFIA   97266  38536227
2961      2962    FERREIRO CEDONES, JAZMIN SOFIA   97266  38536227
4030      4031                   HOLERT, IGNACIO   98393  39469845
4031      4032                   HOLERT, IGNACIO   98393  39469845
4175      4176            IVALDI LEVRINO, FRANCO   94175  36728851
4176      4177            IVALDI LEVRINO, FRANCO   94175  36728851
4201      4202                  JANNELLO, MATIAS   96479  37431588
4202      4203                  JANNELLO, MATIAS   96479  37431588
4509      4510            LEDESMA, DIEGO ARMANDO   94610  34338817
4510      4511            LEDESMA, DIEGO ARMANDO   94610  34338817
4755      4756                     LOYARTE, IVAN   97213  37170681
4756      4757                     LOYARTE, IVAN   97213  37170681
4813      4814                   LUNGWITZ, KLAUS   96416  38010229
4814      4815                   LUNGWITZ, KLAUS   96416  38010229
5765      5766            OBST DEMAESTRI, HERMAN   97884  39757329
5766      5767            OBST DEMAESTRI, HERMAN   97884  39757329
6166      6167  PEREZ BINNEWIES, GUADALUPE AMIRA   96378  37280933
6167      6168  PEREZ BINNEWIES, GUADALUPE AMIRA   96378  37280933
6239      6240                     PESADO, LUCÍA   98275  39245042
6240      6241                     PESADO, LUCÍA   98275  39245042
6318      6319      PINEDA MONTILLA, ELVIS JANED  100452  95521916
6319      6320      PINEDA MONTILLA, ELVIS JANED  100452  95521916
6780      6781                      RIEMER, ALAN   92767  36601118
6781      6782                      RIEMER, ALAN   92767  36601118
7143      7144                 RUDERMAN, GABRIEL   98253  39459056
7144      7145                 RUDERMAN, GABRIEL   98253  39459056

In [168]:
len(cd_17)

8722

In [169]:
cd_17 = cd_17[~cd_17.duplicated(subset="DNI", keep="first")]

In [170]:
len(cd_17)

8704

### CEI

In [171]:
cei_16.duplicated(subset="DNI").sum()

7

In [172]:
cei_16[cei_16.duplicated(subset="DNI", keep=False)]

N° ORDEN                             NOMBRE Padrón       DNI
673        674              ARREJIN, MAURO ALEXIS  97555  35266598
674        675              ARREJIN, MAURO ALEXIS  97555  35266598
715        716             ASTUDILLO, JUAN MARCOS  98118  38426324
716        717             ASTUDILLO, JUAN MARCOS  98118  38426324
2278      2279                     CAUHEPE, LUCAS  98354  39415261
2279      2280                     CAUHEPE, LUCAS  98354  39415261
2838      2839                CRAVERO, JUAN PABLO  98379  38305305
2839      2840                CRAVERO, JUAN PABLO  98379  38305305
3976      3977  FERNÁNDEZ SÁNCHEZ, MARTÍN NICOLÁS  98410  39466298
3977      3978  FERNÁNDEZ SÁNCHEZ, MARTÍN NICOLÁS  98410  39466298
7738      7739                 NISCA, MARIA ABRIL  97736  38939419
7739      7740                 NISCA, MARIA ABRIL  97736  38939419
8211      8212              PAZOS, MATIAS NICOLAS  97657  38617131
8212      8213              PAZOS, MATIAS NICOLAS  97657  38617131

En el padrón del CEI del 2016 hay 7 duplicados. En los otros padrones no. Los saco acá mismo así hacemos las cuentas más fácil.

In [173]:
len(cei_16)

11793

In [174]:
cei_16 = cei_16[~cei_16.duplicated(subset="DNI", keep="first")]

In [175]:
len(cei_16)

11786

Como corresponde, hay 7 menos, que son los duplicados.

## Armado de padrón
### Agregado de CD 2017
Agregamos los del padrón de CD de 2017 a los del CEI 2016. 

In [176]:
cei_17 = cei_16.append(cd_17)

In [177]:
cei_17.isnull().sum()

N° ORDEN    0
NOMBRE      0
Padrón      0
DNI         0
dtype: int64

Todo en orden. Vamos a ver los duplicados.

In [178]:
dup_dni = cei_17[cei_17.duplicated(subset="DNI", keep=False)].sort_values("DNI")
dup_dni

N° ORDEN                            NOMBRE  Padrón       DNI
6723       6724                RESIO, LUIS BENITO   33024  10390436
9088       9089                RESIO, LUIS BENITO   33024  10390436
5572       5573                IBARRA, SARA MABEL   64135  10972945
4084       4085                IBARRA, SARA MABEL   64135  10972945
7393       7394          MONSALVE, GUSTAVO ADOLFO   36561  11122336
5461       5462          MONSALVE, GUSTAVO ADOLFO   36561  11122336
9435       9436             ROJAS, MARCELO ANDRES   38601  11802981
6979       6980             ROJAS, MARCELO ANDRES   38601  11802981
5444       5445         MOLINERO, GUSTAVO EDGARDO   39437  11926253
7369       7370         MOLINERO, GUSTAVO EDGARDO   39437  11926253
7585       7586          SERVIDIO, FRANCISCO LUIS   42107  12088273
10248     10249          SERVIDIO, FRANCISCO LUIS   42107  12088273
937         938              BARRAL, JORGE NESTOR   43655  12865402
696         697              BARRAL, JORGE NESTOR   43655  12865402
6872       6873              RODONI, MABEL SUSANA   48332  12945878
9282       9283              RODONI, MABEL SUSANA   48332  12945878
7163       7164              MENDEZ, RAUL RICARDO   51555  13621375
5297       5298              MENDEZ, RAUL RICARDO   51555  13621375
7794       7795        STEKELORUM, CARLOS ALBERTO   52778  13852547
10543     10544        STEKELORUM, CARLOS ALBERTO   52778  13852547
639         640               ARILLO, LUIS ANIBAL   56078  14391762
481         482               ARILLO, LUIS ANIBAL   56078  14391762
8093       8094         PAOLANTONIO, CARLOS DARIO   51492  14768695
5973       5974         PAOLANTONIO, CARLOS DARIO   51492  14768695
5870       5871              KNAPP, FERNANDA ALBA   54428  16057084
4312       4313              KNAPP, FERNANDA ALBA   54428  16057084
5936       5937           PAILOS, EDGARDO GUSTAVO   69411  16266501
8041       8042           PAILOS, EDGARDO GUSTAVO   69411  16266501
1747       1748            CHAMI, FABIANA MARCELA   54437  16288413
2396       2397            CHAMI, FABIANA MARCELA   54437  16288413
...         ...                               ...     ...       ...
2941       2942      CURBELO FERNANDEZ, ALEJANDRA   99286  95464402
2122       2123      CURBELO FERNANDEZ, ALEJANDRA   99286  95464402
3973       3974       HERNANDEZ MEZA, JOHN FELIPE   99568  95466366
5425       5426       HERNANDEZ MEZA, JOHN FELIPE   99568  95466366
1929       1930      CAÑON BELLO, CRISTIAN FELIPE  100162  95466824
1418       1419      CAÑON BELLO, CRISTIAN FELIPE  100162  95466824
6557       6558  QUIÑONES IZARRA, HECTOR REINALDO   99934  95469713
8870       8871  QUIÑONES IZARRA, HECTOR REINALDO   99934  95469713
2592       2593        COCA ABAD, ANTONY LEONARDO  100335  95479131
1872       1873        COCA ABAD, ANTONY LEONARDO  100335  95479131
8084       8085             PANOSO RUIZ, REINALDO  100147  95494549
5970       5971             PANOSO RUIZ, REINALDO  100147  95494549
6261       6262                PETROVSKIY, GEORGY  100084  95495017
8481       8482                PETROVSKIY, GEORGY  100084  95495017
1692       1693  CCARAHUANCO TOSCANO, LUIS MIGUEL  100098  95498722
2309       2310  CCARAHUANCO TOSCANO, LUIS MIGUEL  100098  95498722
2386       2387        CHACIN AMARAL, MARCO TULIO   99749  95502883
1739       1740        CHACIN AMARAL, MARCO TULIO   99749  95502883
6533       6534      QUEIROZ BEZERRA, JOAO CARLOS  100310  95512258
8840       8841      QUEIROZ BEZERRA, JOAO CARLOS  100310  95512258
9777       9778    SALAZAR BARRIOS, ADRIAN MANUEL  100476  95513966
7241       7242    SALAZAR BARRIOS, ADRIAN MANUEL  100476  95513966
2007       2008    CARDENAS QUISPE, ANDREW MIGUEL  100436  95521898
1477       1478    CARDENAS QUISPE, ANDREW MIGUEL  100436  95521898
8557       8558      PINEDA MONTILLA, ELVIS JANED  100452  95521916
6318       6319      PINEDA MONTILLA, ELVIS JANED  100452  95521916
4555       4556     GARCIA GARAYAR, MARLON DANIEL  100262  9555

Me fijo que no haya dnis duplicados que no sean efectivamente duplicados.

In [179]:
dup_no = cei_17[cei_17.duplicated(subset=["DNI", "Padrón"], keep=False)].sort_values("DNI")
dup_no

N° ORDEN                            NOMBRE  Padrón       DNI
9088       9089                RESIO, LUIS BENITO   33024  10390436
6723       6724                RESIO, LUIS BENITO   33024  10390436
4084       4085                IBARRA, SARA MABEL   64135  10972945
5572       5573                IBARRA, SARA MABEL   64135  10972945
5461       5462          MONSALVE, GUSTAVO ADOLFO   36561  11122336
7393       7394          MONSALVE, GUSTAVO ADOLFO   36561  11122336
9435       9436             ROJAS, MARCELO ANDRES   38601  11802981
6979       6980             ROJAS, MARCELO ANDRES   38601  11802981
5444       5445         MOLINERO, GUSTAVO EDGARDO   39437  11926253
7369       7370         MOLINERO, GUSTAVO EDGARDO   39437  11926253
10248     10249          SERVIDIO, FRANCISCO LUIS   42107  12088273
7585       7586          SERVIDIO, FRANCISCO LUIS   42107  12088273
696         697              BARRAL, JORGE NESTOR   43655  12865402
937         938              BARRAL, JORGE NESTOR   43655  12865402
6872       6873              RODONI, MABEL SUSANA   48332  12945878
9282       9283              RODONI, MABEL SUSANA   48332  12945878
7163       7164              MENDEZ, RAUL RICARDO   51555  13621375
5297       5298              MENDEZ, RAUL RICARDO   51555  13621375
7794       7795        STEKELORUM, CARLOS ALBERTO   52778  13852547
10543     10544        STEKELORUM, CARLOS ALBERTO   52778  13852547
639         640               ARILLO, LUIS ANIBAL   56078  14391762
481         482               ARILLO, LUIS ANIBAL   56078  14391762
8093       8094         PAOLANTONIO, CARLOS DARIO   51492  14768695
5973       5974         PAOLANTONIO, CARLOS DARIO   51492  14768695
4312       4313              KNAPP, FERNANDA ALBA   54428  16057084
5870       5871              KNAPP, FERNANDA ALBA   54428  16057084
8041       8042           PAILOS, EDGARDO GUSTAVO   69411  16266501
5936       5937           PAILOS, EDGARDO GUSTAVO   69411  16266501
1747       1748            CHAMI, FABIANA MARCELA   54437  16288413
2396       2397            CHAMI, FABIANA MARCELA   54437  16288413
...         ...                               ...     ...       ...
2941       2942      CURBELO FERNANDEZ, ALEJANDRA   99286  95464402
2122       2123      CURBELO FERNANDEZ, ALEJANDRA   99286  95464402
3973       3974       HERNANDEZ MEZA, JOHN FELIPE   99568  95466366
5425       5426       HERNANDEZ MEZA, JOHN FELIPE   99568  95466366
1929       1930      CAÑON BELLO, CRISTIAN FELIPE  100162  95466824
1418       1419      CAÑON BELLO, CRISTIAN FELIPE  100162  95466824
6557       6558  QUIÑONES IZARRA, HECTOR REINALDO   99934  95469713
8870       8871  QUIÑONES IZARRA, HECTOR REINALDO   99934  95469713
2592       2593        COCA ABAD, ANTONY LEONARDO  100335  95479131
1872       1873        COCA ABAD, ANTONY LEONARDO  100335  95479131
8084       8085             PANOSO RUIZ, REINALDO  100147  95494549
5970       5971             PANOSO RUIZ, REINALDO  100147  95494549
6261       6262                PETROVSKIY, GEORGY  100084  95495017
8481       8482                PETROVSKIY, GEORGY  100084  95495017
1692       1693  CCARAHUANCO TOSCANO, LUIS MIGUEL  100098  95498722
2309       2310  CCARAHUANCO TOSCANO, LUIS MIGUEL  100098  95498722
2386       2387        CHACIN AMARAL, MARCO TULIO   99749  95502883
1739       1740        CHACIN AMARAL, MARCO TULIO   99749  95502883
6533       6534      QUEIROZ BEZERRA, JOAO CARLOS  100310  95512258
8840       8841      QUEIROZ BEZERRA, JOAO CARLOS  100310  95512258
9777       9778    SALAZAR BARRIOS, ADRIAN MANUEL  100476  95513966
7241       7242    SALAZAR BARRIOS, ADRIAN MANUEL  100476  95513966
2007       2008    CARDENAS QUISPE, ANDREW MIGUEL  100436  95521898
1477       1478    CARDENAS QUISPE, ANDREW MIGUEL  100436  95521898
8557       8558      PINEDA MONTILLA, ELVIS JANED  100452  95521916
6318       6319      PINEDA MONTILLA, ELVIS JANED  100452  95521916
4555       4556     GARCIA GARAYAR, MARLON DANIEL  100262  9555

Hay 1 DNI repetido cuyo padrón no coincide.

In [180]:
dup_dni[~dup_dni.DNI.isin(dup_no.DNI)]

N° ORDEN                                            NOMBRE Padrón  \
9780      9781                    SALAZAR VARGAS, CARLOS ROBERTO  B0239   
7243      7244  SALAZAR VARGAS, CARLOS ROBERTO                 B    239   

           DNI  
9780  93890243  
7243  93890243

Los que coniciden en DNI están efectivamente duplicados. Los quitamos.

In [181]:
cei_17 = cei_17[~cei_17.duplicated(subset="DNI", keep="first")]

In [182]:
cei_17.duplicated(subset="DNI").sum()

0

Duplicados fueron eliminados. Ahora podemos ver cuántos estudiantes se agregaron al padrón.

In [183]:
len(cei_17)-len(cei_16)

1634

### Remoción de Graduados

In [184]:
grad_dni = cei_17[cei_17.DNI.isin(grad_16.DNI)]
len(grad_dni)

196

In [185]:
grad_pad = cei_17[cei_17["Padrón"].isin(grad_16["Padrón"])]
len(grad_pad)

196

In [186]:
grad_dni.DNI.isin(grad_pad.DNI).sum()

196

Confirmado por padrón y DNI, son 196 graduados los que se van. Los quitamos.

In [187]:
cei_17 = cei_17[~cei_17.DNI.isin(grad_16.DNI)]

Cantidad _neta_ agregada, por si sirve de algo.

In [188]:
len(cei_17) - len(cei_16)

1438

## Escribimos el padrón a un csv

Primero ordenamos por apellido y pisamos el número de orden.

In [189]:
cei_17 = cei_17.sort_values("NOMBRE", ascending=True).reset_index().drop(["index", "N° ORDEN"], axis=1)

In [190]:
cei_17 = cei_17.reset_index()

In [191]:
cei_17["index"] = cei_17["index"]+1

In [192]:
cei_17.rename(columns={"index": "N° Orden"}, inplace=True)

In [193]:
cei_17

N° Orden                                   NOMBRE  Padrón        DNI
0             1                       AAS, PETTER FOSSAN  100346   29086232
1             2                           ABABOU, KHALIL   97708  RA1057904
2             3                      ABAD, ARIEL IGNACIO   86213   32166584
3             4                             ABAD, MANUEL   89007   32553674
4             5                            ABADI, BETINA   93386   35902423
5             6                          ABAL, FLORENCIA   98128   39743372
6             7             ABALLAY PIÑEIRO, MARIA BELEN   97420   39321842
7             8                      ABALOS, JOSÉ DANIEL   98584   38748437
8             9                 ABALOS, PATRICIO RICARDO   92850   35970834
9            10                    ABATE, FERNANDO ANGEL   71962   23277085
10           11                    ABATE, MIGUEL AGUSTÍN   84084   31207574
11           12                             ABATE, TOMAS   90894   34630309
12           13               ABAURRE, NICOLAS ALEJANDRO   78208   27356596
13           14                             ABBA, MARCOS   90614   33619728
14           15                          ABBATE, MARIANO  100142   39434276
15           16           ABBENDA DE OTO, FRANCO LUCIANO  101048   40513155
16           17                         ABBENE, FEDERICO   91032   34593429
17           18                    ABBONDANZA, ALEJANDRO   77296   26000949
18           19                          ABBOUD, IGNACIO   96567   38521607
19           20                      ABD, PABLO EZEQUIEL   86745   31223954
20           21                       ABDO, EDUARDO JOSE   93557   36826533
21           22               ABECHIAN, EMILIANO GABRIEL  101648   40731619
22           23                   ABEIJÓN BASSET, FRANCO   98762   39171584
23           24                ABELE, ANTONELLA CATERINA   94512   37432783
24           25                        ABELENDA, FACUNDO   93590   36636391
25           26                  ABELLA, DIEGO ALEJANDRO   84443   31438529
26           27                       ABENTIN, FRANCISCO   79277   27778370
27           28                        ABERASTURY, LUCÍA   93150   36158896
28           29                      ABERASTURY, MARIANA   91087   34744380
29           30                     ABIERI, JUAN AUGUSTO   97483   38886785
...         ...                                      ...     ...        ...
13194     13195               de URTIAGA, RODRIGO ADOLFO   93344   36688572
13195     13196               de VALAIS, EZEQUIEL MARTIN   94463   36882289
13196     13197                   de VEDIA, SEGUNDO JOSE   99720   39334126
13197     13198                           de VERA, SOFÍA  100982   39756599
13198     13199            de ZABALETA CLERICI, GERVASIO   97825   39417045
13199     13200         de la FUENTE BADANO, JUAN MANUEL   89647   34001308
13200     13201                     de la FUENTE, HERNAN   95730   38202215
13201     13202  de la HORRA GATTELLI, NATALIA GUADALUPE   92555   34080467
13202     13203              de la PLATA, FACUNDO GASTON  100558   40132040
13203     13204                        de la RUA, MANUEL   93692   36728382
13204     13205                       de la VEGA, MARTÍN   90665   34434804
13205     13206                    de los HEROS, AGUSTIN  101897   40063610
13206     13207                    de los HEROS, LAUTARO   95414   38300972
13207     13208                  de los HEROS, MARIANELA   93050   35992265
13208     13209            de los SANTOS, RODRIGO MARTÍN   92380   35969203
13209     13210      del CAMPO CARRANZA, CRISTIAN ANDRES   93775   36153564
13210     13211                 del ESTAL, LUCAS GABRIEL   94430   37339421
13211     13212                       del MAZO, FEDERICO  100029   40230465
13212     13213               del MOLINO TORRES, JOAQUIN   99369   39909107
13213     13214              del POZO, FRANCISCO MARCELO  101656   39773918
13214     13215                 del PRADO, JORGE ALBERT

In [194]:
cei_17.to_csv("padron_cei_2017.csv", index=False)

## Padrón exclusivo del CEI
Los que solo votan en CEI y no están en el padrón del CD.

In [195]:
cei_en_cd = cei_17[cei_17.DNI.isin(cd_17.DNI)]
len(cei_en_cd)

8673

In [196]:
cei_en_cd_pad = cei_17[cei_17["Padrón"].isin(cd_17["Padrón"])]
len(cei_en_cd_pad)

8673

In [197]:
cei_en_cd_pad.DNI.isin(cei_en_cd.DNI).sum()

8672

In [198]:
cei_en_cd_pad[~cei_en_cd_pad.DNI.isin(cei_en_cd.DNI)]

N° Orden                           NOMBRE Padrón   DNI
3802      3803  DITO, LEANDRO KAMANGOTO GERALDO  91758  6030

In [199]:
cei_en_cd[~cei_en_cd.DNI.isin(cei_en_cd_pad.DNI)]

N° Orden                          NOMBRE Padrón       DNI
10912     10913  SALAZAR VARGAS, CARLOS ROBERTO  B0239  93890243

Hay uno extra por padroń y uno extra por DNI. Vamos a sacar los dos.

In [205]:
cei_17 = cei_17[~cei_17.DNI.isin(cd_17.DNI)]
cei_17 = cei_17[~cei_17["Padrón"].isin(cd_17["Padrón"])]

In [206]:
len(cei_17)

4550

In [207]:
cei_17.to_csv("cei_exclusivo.csv", index=)

N° Orden                 NOMBRE  Padrón        DNI
0          1     AAS, PETTER FOSSAN  100346   29086232
1          2         ABABOU, KHALIL   97708  RA1057904
3          4           ABAD, MANUEL   89007   32553674
9         10  ABATE, FERNANDO ANGEL   71962   23277085
10        11  ABATE, MIGUEL AGUSTÍN   84084   31207574

## Casos raros

In [202]:
padrones_no_numericos = pd.to_numeric(cei_16["Padrón"], errors="force").isnull()

In [203]:
cei_16[padrones_no_numericos]

N° ORDEN                          NOMBRE Padrón       DNI
9780      9781  SALAZAR VARGAS, CARLOS ROBERTO  B0239  93890243

In [204]:
cd_17[cd_17["Padrón"].str.len() < 5]

N° ORDEN                                            NOMBRE Padrón  \
629        630  BALDOCEDA HUALLPA, MAXIMO                      B    204   
7243      7244  SALAZAR VARGAS, CARLOS ROBERTO                 B    239   
8260      8261  VARGAS PÉREZ, VILMA JUANA                      B    206   

           DNI  
629   92859860  
7243  93890243  
8260  92892429